demo/scripts/openseadragon-imaginghelper.js

```diff
@@ -693,8 +693,22 @@
      **/
     function onOpen() {
         this._haveImage = true;
-        this.imgWidth = this._viewer.viewport.contentSize.x;
-        this.imgHeight = this._viewer.viewport.contentSize.y;
+        var contentSizeViewport = {},
+          contentSize;
+        // use world if we can't use contentSize
+        if (!this._viewer.viewport.contentSize){
+            var homeBounds = this._viewer.world.getHomeBounds();
+            contentSizeViewport.x = homeBounds.width - homeBounds.x;
```

Comment:
I'm kind of new to this code; can you explain why you're subtracting the x from the width here?